### PR TITLE
fix(core): fix checkbox transition from indeterminate to false state

### LINKED ISSUE #498: [BUG] Checkbox displays tick on transition from indeterminate to false state
# 🐞 Bug report

### Description

When I'm changing checkbox state from `null` to `false`, for a split second checkbox displays tick like it's in `true` state.

### Reproduction

See reproduction here:
https://stackblitz.com/edit/tui-checkbox-null-to-false?file=src/app/app.component.ts

### Expected behavior

When changing checkbox state from `null` to `false` it shouldn't display `true` state.

### Versions

-   OS: Win 10
-   Chrome 91.0.4472.106
-   Angular 12.0.4
-   taiga-ui/kit@2.11.2

### PR DESCRIPTION
**Pull request by vladimirpotekhin**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [ ] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0-beta.4/)
-   [ ] Tests for the changes have been added (for bug fixes / features)
-   [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

-   [x] Bugfix
-   [ ] Feature
-   [ ] Refactoring (no functional changes, no api changes)
-   [ ] Other... Please describe:

## What is the current behavior?

Closes #498 

## Does this PR introduce a breaking change?

-   [ ] Yes
-   [x] No

### CODE REVIEW

File: projects/core/components/primitive-checkbox/primitive-checkbox.component.ts [hunk 1]
```diff
@@ -67,9 +69,7 @@ export class TuiPrimitiveCheckboxComponent {
     }
 
     get icon(): PolymorpheusContent<TuiContextWithImplicit<TuiSizeL>> {
-        return this.value === null
-            ? this.options.icons.indeterminate
-            : this.options.icons.checked;
+        return this.computedIcon(this.value);
```
Comment (waterplea) [inline projects/core/components/primitive-checkbox/primitive-checkbox.component.ts:72]: Getter updates the state. Not good :(

Comment (vladimirpotekhin) [reply projects/core/components/primitive-checkbox/primitive-checkbox.component.ts]: You're right( I moved icon setting into the value setter. What do you think about that?

Comment (waterplea) [reply projects/core/components/primitive-checkbox/primitive-checkbox.component.ts]: Ok, let's leave it like that, except `icon` getter is unnecessary now, just change `currentIcon` to `icon` and make it public.

Comment (vladimirpotekhin) [reply projects/core/components/primitive-checkbox/primitive-checkbox.component.ts]: ok, done. I also added initial icon value, for smooth first animation

**Review by waterplea — APPROVED** [review]

**Pull request merged**